### PR TITLE
Fix some errors caused by updating Axis xscale or yscale

### LINKED ISSUE #3061: hvlines bug in GLMakie interactive window
There appears to be a bug when trying to reproduce the example in the docs for a list of `xmax`s or `xmin`s. The stack trace seems similar to a bug I reported last year ([discourse post here](https://discourse.julialang.org/t/hlines-xmin-xmax-do-not-accept-observable-list-in-glmakie/90788), [issue here](https://github.com/MakieOrg/Makie.jl/pull/2446)), so perhaps that fix did not address the underlying issue.

I will reproduce the [example from the docs](https://docs.makie.org/stable/examples/plotting_functions/hvlines/) below, but using GLMakie instead of CairoMakie. (CairoMakie works)
```
using GLMakie
GLMakie.activate!(inline=false)

f = Figure()

ax1 = Axis(f[1, 1], title = "vlines")

lines!(ax1, 0..4pi, sin)
vlines!(ax1, [pi, 2pi, 3pi], color = :red)

ax2 = Axis(f[1, 2], title = "hlines")
hlines!(ax2, [1, 2, 3, 4], xmax = [0.25, 0.5, 0.75, 1], color = :blue)

f
```

Output from fresh session:
```
┌ Warning: Can't convert Observable{Vector{Float64}} to opengl, for attribute xmax
└ @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLTypes.jl:432
ERROR: MethodError: no method matching gl_convert(::Vector{Float64})

Closest candidates are:
  gl_convert(::T) where T<:ColorTypes.Colorant
   @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLUniforms.jl:194
  gl_convert(::T) where T<:GeometryBasics.Mesh
   @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLUniforms.jl:196
  gl_convert(::T) where T<:GeometryBasics.AbstractMesh
   @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLUniforms.jl:195
  ...

Stacktrace:
  [1] #map#13
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:564 [inlined]
  [2] map
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:562 [inlined]
  [3] const_lift
    @ ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLUtils.jl:106 [inlined]
  [4] gl_convert(s::Observable{Vector{Float64}})
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLUniforms.jl:228
  [5] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window, context::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLTypes.jl:430
  [6] GLMakie.GLAbstraction.RenderObject(data::Dict{Symbol, Any}, program::GLMakie.GLVisualizeShader, pre::GLMakie.GLAbstraction.StandardPrerender, post::GLFW.Window)
    @ GLMakie.GLAbstraction ~/.julia/packages/GLMakie/OVxAi/src/GLAbstraction/GLTypes.jl:404
  [7] assemble_shader(data::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/glshaders/visualize_interface.jl:108
  [8] draw_linesegments(screen::Any, positions::Observable{Vector{Point{2, Float32}}}, data::Dict)
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/glshaders/lines.jl:156
  [9] (::GLMakie.var"#209#210"{GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}})(gl_attributes::Dict{Symbol, Any})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:334
 [10] (::GLMakie.var"#175#178"{GLMakie.var"#209#210"{GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}})()
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:111
 [11] get!(default::GLMakie.var"#175#178"{GLMakie.var"#209#210"{GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}}, GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}}, h::Dict{UInt64, GLMakie.GLAbstraction.RenderObject}, key::UInt64)
    @ Base ./dict.jl:468
 [12] cached_robj!(robj_func::GLMakie.var"#209#210"{GLMakie.Screen{GLFW.Window}, Scene, LineSegments{Tuple{Vector{Point{2, Float32}}}}}, screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::LineSegments{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:84
 [13] draw_atomic
    @ ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:310 [inlined]
 [14] insert!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::LineSegments{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:126
 [15] (::GLMakie.var"#181#182"{GLMakie.Screen{GLFW.Window}, Scene})(x::LineSegments{Tuple{Vector{Point{2, Float32}}}})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:131
 [16] foreach(f::GLMakie.var"#181#182"{GLMakie.Screen{GLFW.Window}, Scene}, itr::Vector{AbstractPlot})
    @ Base ./abstractarray.jl:3073
 [17] insert!(screen::GLMakie.Screen{GLFW.Window}, scene::Scene, x::Combined{Makie.hlines, Tuple{Vector{Int64}}})
    @ GLMakie ~/.julia/packages/GLMakie/OVxAi/src/drawing_primitives.jl:128
 [18] push!(scene::Scene, plot::Combined{Makie.hlines, Tuple{Vector{Int64}}})
    @ Makie ~/.julia/packages/Makie/iECbF/src/scenes.jl:460
 [19] plot!(scene::Scene, P::Type{Combined{Makie.hlines, Tuple{Vector{Int64}}}}, attributes::Attributes, input::Tuple{Observable{Vector{Int64}}}, args::Observable{Tuple{Vector{Int64}}})
    @ Makie ~/.julia/packages/Makie/iECbF/src/interfaces.jl:433
 [20] plot!(scene::Scene, P::Type{Combined{Makie.hlines}}, attributes::Attributes, args::Vector{Int64}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/iECbF/src/interfaces.jl:344
 [21] plot!
    @ ~/.julia/packages/Makie/iECbF/src/interfaces.jl:311 [inlined]
 [22] plot!(la::Axis, P::Type{Combined{Makie.hlines}}, attributes::Attributes, args::Vector{Int64}; kw_attributes::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Makie ~/.julia/packages/Makie/iECbF/src/makielayout/blocks/axis.jl:792
 [23] plot!
    @ ~/.julia/packages/Makie/iECbF/src/makielayout/blocks/axis.jl:779 [inlined]
 [24] #plot!#1405
    @ ~/.julia/packages/Makie/iECbF/src/makielayout/blocks/axis.jl:809 [inlined]
 [25] hlines!(::Axis, ::Vararg{Any}; attributes::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:xmax, :color), Tuple{Vector{Float64}, Symbol}}})
    @ Makie ~/.julia/packages/MakieCore/6sckc/src/recipes.jl:38
 [26] top-level scope
    @ REPL[33]:1
```

- [x] are you running newest version (version from docs) ?
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [x] What platform + GPU are you on?
M1 Mac

### LINKED ISSUE #3065: vlines! throws error when changing yscale from log10 back to identity
if i plot a line, and switch yscale to log10 and then back to identity, no problems:

```
julia> using GLMakie

julia> fig, ax, li = lines(1:10, 1:10)

julia> ax.yscale = log10
log10 (generic function with 11 methods)

julia> ax.yscale = identity
identity (generic function with 1 method)
```

but if i then add a vline, initially it's fine with identity, and also after switching to log10, but then it throws a domain error when reverting back to identity:

```
julia> vlines!(ax, 3)
Combined{Makie.vlines, Tuple{Int64}}

julia> ax.yscale = log10
log10 (generic function with 11 methods)

julia> ax.yscale = identity
ERROR: DomainError with -0.25963733:
log10 will only return a complex result if called with a complex argument. Try log10(Complex(x)).
Stacktrace:
  [1] throw_complex_domainerror(f::Symbol, x::Float32)
    @ Base.Math ./math.jl:33
  [2] _log(x::Float32, base::Val{10}, func::Symbol)
    @ Base.Math ./special/log.jl:336
  [3] log10
    @ ./special/log.jl:265 [inlined]
  [4] apply_transform
    @ ~/.julia/packages/Makie/iECbF/src/layouting/transformation.jl:275 [inlined]
  [5] #767
    @ ~/.julia/packages/Makie/iECbF/src/layouting/transformation.jl:271 [inlined]
  [6] iterate
    @ ./generator.jl:47 [inlined]
  [7] _collect(c::Vector{Point{2, Float32}}, itr::Base.Generator{Vector{Point{2, Float32}}, Makie.var"#767#768"{Tuple{typeof(identity), typeof(log10)}}}, #unused#::Base.EltypeUnknown, isz::Base.HasShape{1})
    @ Base ./array.jl:802
  [8] collect_similar
    @ ./array.jl:711 [inlined]
  [9] map
    @ ./abstractarray.jl:3261 [inlined]
 [10] apply_transform
    @ ~/.julia/packages/Makie/iECbF/src/layouting/transformation.jl:271 [inlined]
 [11] apply_transform
    @ ~/.julia/packages/Makie/iECbF/src/layouting/transformation.jl:213 [inlined]
 [12] (::Makie.var"#763#764")(f::Tuple{typeof(identity), typeof(log10)}, d::Vector{Point{2, Float32}}, s::Symbol)
    @ Makie ~/.julia/packages/Makie/iECbF/src/layouting/transformation.jl:215
 [13] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:816
 [14] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:813
 [15] (::Observables.MapCallback)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
 [16] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [17] invokelatest
    @ ./essentials.jl:813 [inlined]
 [18] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [19] setindex!(observable::Observable, val::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
--- the last 5 lines are repeated 1 more time ---
 [25] (::Makie.var"#175#177"{Attributes, Observable{Tuple{Vector{Point{2, Float32}}}}, DataType})(kwargs::Tuple{}, args::Vector{Point{2, Float32}})
    @ Makie ~/.julia/packages/Makie/iECbF/src/interfaces.jl:342
 [26] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:816
 [27] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:813
 [28] (::Observables.OnAny)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [29] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [30] invokelatest
    @ ./essentials.jl:813 [inlined]
 [31] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [32] (::Makie.var"#577#579"{Combined{Makie.vlines, Tuple{Int64}}, Observable{Vector{Point{2, Float32}}}})(lims::GeometryBasics.HyperRectangle{2, Float32}, vals::Int64, mi::Int64, ma::Int64, transf::Tuple{typeof(identity), typeof(identity)})
    @ Makie ~/.julia/packages/Makie/iECbF/src/basic_recipes/hvlines.jl:81
 [33] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:816
 [34] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:813
 [35] (::Observables.OnAny)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [36] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [37] invokelatest
    @ ./essentials.jl:813 [inlined]
 [38] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [39] setindex!(observable::Observable, val::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [40] (::Observables.MapCallback)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:431
--- the last 5 lines are repeated 1 more time ---
 [46] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [47] invokelatest
    @ ./essentials.jl:813 [inlined]
 [48] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [49] setindex!(observable::Observable, val::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [50] set_proj_view!
    @ ~/.julia/packages/Makie/iECbF/src/camera/camera.jl:96 [inlined]
 [51] update_axis_camera(camera::Camera, t::Tuple{typeof(identity), typeof(identity)}, lims::GeometryBasics.HyperRectangle{2, Float32}, xrev::Bool, yrev::Bool)
    @ Makie ~/.julia/packages/Makie/iECbF/src/makielayout/blocks/axis.jl:93
 [52] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:816
 [53] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:813
 [54] (::Observables.OnAny)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [55] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [56] invokelatest
    @ ./essentials.jl:813 [inlined]
 [57] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [58] setindex!(observable::Observable, val::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [59] (::Makie.var"#1321#1351"{Scene})(xsc::Function, ysc::Function)
    @ Makie ~/.julia/packages/Makie/iECbF/src/makielayout/blocks/axis.jl:251
 [60] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base ./essentials.jl:816
 [61] invokelatest(::Any, ::Any, ::Vararg{Any})
    @ Base ./essentials.jl:813
 [62] (::Observables.OnAny)(value::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:415
 [63] #invokelatest#2
    @ ./essentials.jl:816 [inlined]
 [64] invokelatest
    @ ./essentials.jl:813 [inlined]
 [65] notify
    @ ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:169 [inlined]
 [66] setindex!(observable::Observable, val::Any)
    @ Observables ~/.julia/packages/Observables/PHGQ8/src/Observables.jl:86
 [67] setproperty!(x::Axis, key::Symbol, value::Function)
    @ Makie ~/.julia/packages/Makie/iECbF/src/makielayout/blocks.jl:445

julia> VERSION
v"1.9.2"

(jl_SW45sZ) pkg> st
Status `/private/var/folders/s5/8d629n5d7nsf37f60_91wzr40000gq/T/jl_SW45sZ/Project.toml`
  [e9467ef8] GLMakie v0.8.6

julia> Sys.MACHINE
"arm64-apple-darwin22.4.0"
```

### PR DESCRIPTION
**Pull request by ffreyer**
# Description

<details>
<summary>Old Text</summary>

This should make the default behavior that transform_func is passed down to child scenes and plots first, before calculations happen. 

This fixes #3065, but I haven't tested if it breaks anything else yet.

Side note - Why doesn't `map` accept `priority`? And copy, given that it links to the original?
</details>

This pr tweaks some behavior around axis scales/transform_funcs to avoid mostly limit related errors.

Changes:
- Added `priority` kwarg to our map, map! and lift implementations (Observables is still missing this)
- Adjusted the order of updates when changing `ax.xscale` or `ax.yscale`. This allows barplot to fix its bars before the axis checks them for limits. Basically it makes #3004 work with dynamic transform_func/scale changes.
    - previously: update limits first, then `ax.scene.transformation.transform_func`, then axis camera matrices, then plots
    - now: update `ax.scene.transformation.transform_func` first, then plots, then limits, then axis camera (order enforced by priority)
- `Transformation(parent)` now has an overwrite for `transform_func`
- `hvlines` and `hvspan` no longer back-transform their relative limits to input space. Instead they apply their transform_func to their input data and remove it from their child plots. So rather than plotting to input space they now plot to transformed space. This fixes #3065
- `barplot` now runs `calculate_bars!` at priority 1 to make sure it runs before transform_func is applied in its child plot(s). This shouldn't be necessary, but I left it in for now. 
- `errorbar` and `rangebar` changed from using an array of pairs to just a flat array for its linesegments. This avoid a conversion error from `Vector` to `ReinterpretedArray`.
- remove xmin, xmax, ymin, ymax, xautolimits, yautolimits from attribute passthrough in hvlines and hvspan. fixes #3061

I have tested these changes with
```julia
using GLMakie
begin
    fig, ax, li = lines(1:10, 1:10)
    vlines!(ax, 3)
    hlines!(ax, 3)
    bp = barplot!(ax, 1 .+ 5 .* rand(10))
    vspan!(ax, 3, 4)
    hspan!(ax, 3, 4)
    bracket!(ax, 1, 1, 2, 2)
    eb = errorbars!(ax, 1:10, 1:10, [0.3 for _ in 1:10], whiskerwidth = 5)
    text!(ax, Point2f(2), text = "abba")
    tooltip!(ax, Point2f(8), "baab")
    tricontourf!(ax, 1 .+ 4 .* rand(5), 1 .+ 4 .* rand(5), rand(5))
    qqplot!(ax, 5:10, 1:5)
    display(fig)
end
ax.yscale = log10
ax.yscale = identity
```

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Added an entry in NEWS.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [x] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

**Comment by ffreyer** [comment]
`barplot!` (still?) behaves badly with this. Everything else that directly involves `transform_func` seems fine to me.

Alternatively to this we could also lower the priority of limit updates in axis etc. Not sure what's better here.

File: src/basic_recipes/hvspan.jl [hunk 1]
```diff
@@ -74,7 +73,12 @@ function Makie.plot!(p::Union{HSpan, VSpan})
 
     notify(p[1])
 
-    poly!(p, rects; p.attributes...)
+    poly_attributes = copy(p.attributes)
+    delete!.(poly_attributes, (:ymin, :ymax, :xmin, :xmax, :xautolimits, :yautolimits))
```
Comment (SimonDanisch) [inline src/basic_recipes/hvspan.jl:77]: better use foreach if you dont want to create a vector!

**Comment by SimonDanisch** [comment]
@jkrumbiegel @ffreyer can we merge this?

**Comment by ffreyer** [comment]
This was ready on my side. I thought that Julius might have some edge cases in mind related to the changes in the update order in Axis. If not we can go ahead and merge I'd say.

**Review by SimonDanisch — APPROVED** [review]

**Pull request merged**